torchdynamo/utils.py

```diff
@@ -245,28 +258,31 @@ def clone_input(x):
             (x.data_ptr() - buffer.data_ptr()) % 32
         ) // x.element_size()
         result = torch.as_strided(buffer, x.size(), x.stride(), cache_line_offset)
+        set_tensor_attributes(result, x, is_train)
         try:
             result.copy_(x.clone())
         except RuntimeError:
             # RuntimeError: unsupported operation: more than one element of the written-to
             # tensor refers to a single memory location. Please clone() the tensor before
             # performing the operation.
-            return torch.clone(x)
+            result = torch.clone(x)
+            set_tensor_attributes(result, x, is_train)
```

Comment:
`torch.clone` copies the requires_grad field as well. So, no need to explicitly set this.

Reply:
Ok.